pkg/localmetrics/localmetrics.go

```diff
@@ -181,33 +181,27 @@ func AddCertificateIssuance(action string) {
 
 // UpdateCertValidDuration updates the Prometheus metric for certificate validity duration.
 func UpdateCertValidDuration(kubeClient client.Client, cert *x509.Certificate, now time.Time, clusterName, namespace string) {
-	if kubeClient == nil {
-		// If kubeClient is nil, set the metric value based on the certificate's expiration date
-		if cert != nil {
-			diff := cert.NotAfter.Sub(now)
-			days := math.Max(0, math.Round(diff.Hours()/24))
-			MetricCertValidDuration.With(prometheus.Labels{
-				"cn":      cert.Subject.CommonName,
-				"cluster": clusterName,
-			}).Set(days)
-		} else {
-			MetricCertValidDuration.With(prometheus.Labels{
-				"cn":      clusterName,
-				"cluster": clusterName,
-			}).Set(0)
-		}
+	var days float64
+	var cn string
+
+	// If both kubeClient and cert are nil, set to 0
+	if kubeClient == nil && cert == nil {
+		MetricCertValidDuration.With(prometheus.Labels{
+			"cn":      clusterName,
+			"cluster": clusterName,
+		}).Set(0)
 		return
 	}
```

Comment:
Do we actually need this conditional? If I'm reading this logic correctly (and maybe I'm not), this same case is already covered below.